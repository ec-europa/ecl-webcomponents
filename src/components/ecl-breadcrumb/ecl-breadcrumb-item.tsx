import { Component, h, Prop} from '@stencil/core';

@Component({
  tag: 'ecl-breadcrumb-item',
  shadow: false,
  scoped: true,
})
export class EclBreadcrumbItem {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() variant: string;
  @Prop() path: string;
  @Prop() currentPage: boolean = false;
  @Prop() ellipsis: boolean = false;
  @Prop() buttonAriaLabel: string = '';

  getClass(): string {
    const styleClasses = [
      `ecl-breadcrumb__segment`,
      `sc-ecl-breadcrumb-${this.theme}`,
      this.styleClass
    ];

    if (this.ellipsis) {
      styleClasses.push('ecl-breadcrumb__segment--ellipsis');
    }

    return styleClasses.join(' ');
  }

  getLinkClass(): string {
    const linkClasses = [
      `sc-ecl-breadcrumb-${this.theme}`,
      'ecl-breadcrumb__link',
      'ecl-link--no-visited',
    ];

    if (this.variant === 'negative') {
      linkClasses.push('ecl-link--negative');
    }

    return linkClasses.join(' ');
  }

  getLiAttrs() {
    const attrs = { 'data-ecl-breadcrumb-item' : 'static' };
    if (this.ellipsis) {
      attrs['data-ecl-breadcrumb-ellipsis'] = 'data-ecl-breadcrumb-ellipsis';
    }

    return attrs;
  }

  render() {
    return (
      <li
        class={this.getClass()}
        {...this.getLiAttrs()}
      >
      { !this.currentPage && !this.ellipsis ? 
        <ecl-link
          variant="standalone"
          path={this.path}
          style-class={this.getLinkClass()}
        >
          <ecl-icon
            slot="icon-after"
            style-class={`ecl-breadcrumb__icon sc-ecl-breadcrumb-${this.theme}`}
            icon="corner-arrow"
            transform="rotate-90"
            size="2xs"
          ></ecl-icon>
          <slot></slot>
        </ecl-link> :
        <slot></slot>
      }
      { this.ellipsis ? 
        <ecl-button
          style-class={`ecl-breadcrumb__ellipsis sc-ecl-breadcrumb-${this.theme}`}
          variant="ghost"
          data-ecl-breadcrumb-ellipsis-button
          aria-label={this.buttonAriaLabel}
        >
          ...
          <ecl-icon
            style-class={`ecl-breadcrumb__icon sc-ecl-breadcrumb-${this.theme}`}
            size="2xs"
            transform="rotate-90"
            icon="corner-arrow"
            slot="icon-after"
          ></ecl-icon>
        </ecl-button> : '' 
      }
      { this.currentPage ?
        <slot></slot> : ''
      }
      </li>
    );
  }
}