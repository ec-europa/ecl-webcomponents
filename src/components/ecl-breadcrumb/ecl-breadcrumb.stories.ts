const getArgs = () => {
  return {
    variant: 'negative',
  };
};

const getArgTypes = () => {
  return {
    variant: {
      type: { name: 'select' },
      options: ['default', 'negative'],
      description: "Breadcrumb variant",
    },
  };
};

export default {
  title: 'Components/Navigation/breadcrumb',
};

const Template = (args) => 
  `<ecl-breadcrumb
    variant="${args.variant}"
    theme="${args.theme}"
    ecl-script
   >
    <ecl-breadcrumb-item
      path="/example.html"
      theme="${args.theme}"
    >
      Home
    </ecl-breadcrumb-item>
    <ecl-breadcrumb-item
      ellipsis
      button-aria-label="Click to expand"
     >
     </ecl-breadcrumb-item>
    <ecl-breadcrumb-item
      path="/example.html"
      theme="${args.theme}"
    >
      About the European Commission
    </ecl-breadcrumb-item>
    <ecl-breadcrumb-item
      path="/example.html"
      theme="${args.theme}"
    >
      Organisational structure
    </ecl-breadcrumb-item>
    <ecl-breadcrumb-item
      path="/example.html"
      theme="${args.theme}"
    >
      How the Commission is organised
    </ecl-breadcrumb-item>
    <ecl-breadcrumb-item
      current-page
      theme="${args.theme}"
    >
      News
    </ecl-breadcrumb-item>
  </ecl-breadcrumb>`;

export const Breadcrumb = Template.bind({});
Breadcrumb.storyName = 'default';
Breadcrumb.args = getArgs();
Breadcrumb.argTypes = getArgTypes();