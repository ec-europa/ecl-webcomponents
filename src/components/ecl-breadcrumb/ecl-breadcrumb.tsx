import { Component, Prop, h, Element } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";
declare const ECL: any;

@Component({
  tag: 'ecl-breadcrumb',
  styleUrls: {
    ec: './build/styles/ecl-breadcrumb-ec.css',
    eu: './build/styles/ecl-breadcrumb-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build']
})

export class EclBreadcrumb {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() variant: string = 'default';
  @Prop() theme: string = 'ec';
  @Prop() eclScript: boolean = false;
  @Prop() minItemsRight: number = 2;


  getClass(): string {
    return [
      `ecl-breadcrumb`,
      `ecl-breadcrumb--${this.variant}`,
      this.styleClass
    ].join(' ');
  }

  componentDidRender() {
    const items = this.el.querySelectorAll('.ecl-breadcrumb__segment');
    const itemsCount = items.length;
    const ellipsis = this.el.querySelector('.ecl-breadcrumb__segment--ellipsis');
    const startHidingIndex = Array.from(items).indexOf(ellipsis);
    const toBeHidden = Array.from(items).slice((startHidingIndex + 1), (itemsCount - this.minItemsRight));

    toBeHidden.forEach((item) => {
      item.setAttribute('data-ecl-breadcrumb-item', 'expandable');
    });

    this.el.querySelector('.ecl-breadcrumb__container').innerHTML = '';
    this.el.querySelector('.ecl-breadcrumb__container').append(...items);

    if (this.eclScript) {
      // Load the ECL vanilla js if not already present.
      const src = getAssetPath('./build/scripts/ecl-breadcrumb-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const breadcrumb = new ECL.Breadcrumb(this.el.firstElementChild);
        breadcrumb.init();
      };

      document.body.appendChild(script);
    }
  }

  render() {
    return (
      <nav
        class={this.getClass()}
        data-ecl-breadcrumb
      >
        <ol class="ecl-breadcrumb__container">
          <slot></slot>
        </ol>
      </nav>
    )
  }
}

