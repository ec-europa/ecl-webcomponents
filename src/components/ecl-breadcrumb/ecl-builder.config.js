const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, './node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-breadcrumb/breadcrumb.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-breadcrumb-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-breadcrumb-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-breadcrumb-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-breadcrumb-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-breadcrumb-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
