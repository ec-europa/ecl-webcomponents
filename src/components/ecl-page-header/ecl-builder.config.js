const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-page-header-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-page-header-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-page-header-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-page-header-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
  ],
};
