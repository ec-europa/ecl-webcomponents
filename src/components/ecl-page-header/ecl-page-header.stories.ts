const getArgTypes = () => {
  return {
    variant: {
      type: { name: 'select' },
      options: ['default', 'negative'],
      description: 'Page header variant',
    },
    image: {
      type: { name: 'string' },
      description: 'Main page header image',
    },
    imageAlt: {
      type: { name: 'string' },
      description: 'Image alt attribute',
    },
    title: {
      type: { name: 'string' },
      description: 'Title of the header',
    },
    description: {
      type: { name: 'string' },
      description: 'Description',
    },
    meta: {
      type: { name: 'string' },
      description: 'Meta',
    },
    thumbnail: {
      type: { name: 'string' },
      description: 'for the harm',
    },
    thumbnailAlt: {
      type: { name: 'string' },
      description: 'Alt attribute for the thumbnail',
    },
  };
};

export default {
  title: 'Components/Site-wide/page-header',
};

const Template = args =>
  `<ecl-page-header
    variant="${args.variant}"
    theme="${args.theme}"
    header-title="${args.title}"
    image="${args.image || ''}"
    thumbnail="${args.thumbnail}"
    thumbnail-alt="${args.thumbnailAlt}"
    meta="${args.meta}"
    image-alt="${args.imageAlt}"
  >
    <ecl-breadcrumb
      slot="breadcrumb"
      variant="${args.theme === 'ec' ? args.breadcrumbVariant : 'default'}"
      theme="${args.theme}"
      ecl-script
     >
      <ecl-breadcrumb-item
        path="/example.html"
        theme="${args.theme}"
      >
        Home
      </ecl-breadcrumb-item>
      <ecl-breadcrumb-item
        theme="${args.theme}"
        ellipsis
        button-aria-label="Click to expand"
       >
       </ecl-breadcrumb-item>
      <ecl-breadcrumb-item
        path="/example.html"
        theme="${args.theme}"
      >
        About the European Commission
      </ecl-breadcrumb-item>
      <ecl-breadcrumb-item
        path="/example.html"
        theme="${args.theme}"
      >
        Organisational structure
      </ecl-breadcrumb-item>
      <ecl-breadcrumb-item
        path="/example.html"
        theme="${args.theme}"
      >
        How the Commission is organised
      </ecl-breadcrumb-item>
      <ecl-breadcrumb-item
        current-page
        theme="${args.theme}"
      >
        News
      </ecl-breadcrumb-item>
    </ecl-breadcrumb>
    ${args.description}
  </ecl-page-header>`;

export const PageHeaderCore = Template.bind({});
PageHeaderCore.storyName = 'core';
PageHeaderCore.args = {
  variant: 'negative',
  title: 'Page title',
  description: `Lorem ipsum dolor sit amet, <a class="ecl-link" href="/example.html">consectetur adipiscing elit</a>.
                Quisque nec ullamcorper mi. Morbi interdum fermentum tempus. Nam nec rhoncus risus,
                <a class="ecl-link" href="/example.html">eget dictum elit</a>. Vestibulum gravida tincidunt venenatis.`,
  image: 'https://inno-ecl.s3.amazonaws.com/media/examples/example-image2.jpg',
  thumbnail: 'https://inno-ecl.s3.amazonaws.com/media/examples/example-image-square.jpg',
  imageAlt: 'image alternative text',
  meta: 'Meta info | DD Month YYYY',
  thumbnailAlt: 'Europe map',
  breadcrumbVariant: 'negative',
};
PageHeaderCore.argTypes = getArgTypes();

export const PageHeaderHarmonised = Template.bind({});
PageHeaderHarmonised.storyName = 'harmonised';
PageHeaderHarmonised.args = {
  title: 'Page title',
  description: `Lorem ipsum dolor sit amet, <ecl-link path="/example.html">consectetur adipiscing elit</ecl-link>.
                Quisque nec ullamcorper mi. Morbi interdum fermentum tempus. Nam nec rhoncus risus,
                <ecl-link path="/example.html">eget dictum elit</ecl-link>. Vestibulum gravida tincidunt venenatis.`,
  thumbnail: 'https://inno-ecl.s3.amazonaws.com/media/examples/example-image2.jpg',
  meta: 'Meta info | DD Month YYYY',
  breadcrumbVariant: 'default',
}
PageHeaderHarmonised.argTypes = getArgTypes();

export const PageHeaderStandardised = Template.bind({});
PageHeaderStandardised.storyName = 'standardised';
PageHeaderStandardised.args = {
  title: 'Page title',
  description: `Lorem ipsum dolor sit amet, <a class="ecl-link" href="/example.html">consectetur adipiscing elit</a>.
                Quisque nec ullamcorper mi. Morbi interdum fermentum tempus. Nam nec rhoncus risus,
                <a class="ecl-link" href="/example.html">eget dictum elit</a>. Vestibulum gravida tincidunt venenatis.`,
  thumbnail: 'https://inno-ecl.s3.amazonaws.com/media/examples/example-image2.jpg',
  meta: 'Meta info | DD Month YYYY',
  image: 'https://inno-ecl.s3.amazonaws.com/media/examples/example-image-square.jpg',
  imageAlt: 'Alternative text for the image',
  breadcrumbVariant: 'default',
  thumbnailAlt: 'Thumbnail alt attribute',
}
PageHeaderStandardised.argTypes = getArgTypes();
