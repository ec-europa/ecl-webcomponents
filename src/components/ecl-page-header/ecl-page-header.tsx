import { Component, Prop, h, Element } from '@stencil/core';

@Component({
  tag: 'ecl-page-header',
  styleUrls: {
    ec: './build/styles/ecl-page-header-ec.css',
    eu: './build/styles/ecl-page-header-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclPageHeader {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() variant: string = 'default';
  @Prop() theme: string = 'ec';
  @Prop() image: string ;
  @Prop() imageAlt: string;
  @Prop() meta: string;
  @Prop() headerTitle: string;
  @Prop() thumbnail: string;
  @Prop() thumbnailAlt: string;
  @Prop() overlay: string;

  getClass(): string {
    const styleClasses = [
      `ecl-page-header`,
      `ecl-page-header--${this.variant}`,
      this.styleClass
    ]

    if (this.image) {
      styleClasses.push('ecl-page-header--image');
    }

    if (this.overlay) {
      styleClasses.push(`ecl-page-header--overlay-${this.overlay}`);
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    if (this.el.querySelector('.ecl-breadcrumb')) {
      this.el.querySelector('.ecl-breadcrumb').classList.add(`ecl-page-header__breadcrumb`, `sc-ecl-page-header-${this.theme}`);
      this.el.querySelector('.ecl-breadcrumb__container').classList.add(`sc-ecl-page-header-${this.theme}`);
      this.el.querySelectorAll('.ecl-breadcrumb__segment').forEach((segment) => {
        segment.classList.add(`sc-ecl-page-header-${this.theme}`);
      })
    }
  };

  render() {
    return (
      <div class={this.getClass()}>
      { this.image ?
        <ecl-picture
          styleClass={`ecl-page-header__picture-background sc-ecl-page-header-${this.theme}`}
          image={this.image}
          imgClass={`ecl-page-header__background sc-ecl-page-header-${this.theme}`}
        >
          <slot name="sources"></slot>
        </ecl-picture> : ''
      }
        <div class="ecl-container">
          <slot name="breadcrumb"></slot>
        { this.meta ? 
          <div class="ecl-page-header__meta">
            <span class="ecl-page-header__meta-item">{this.meta}</span>
          </div> : ''
        }
          <div class="ecl-page-header__title-container">
            <h1 class="ecl-page-header__title">{this.headerTitle}</h1>
          </div>
          <div class="ecl-page-header__description-container">
          { this.thumbnail ?
            <img
              class="ecl-page-header__description-thumbnail"
              src={this.thumbnail}
              alt={this.thumbnailAlt}
            /> : ''
          }
            <p class="ecl-page-header__description">
              <slot></slot>
            </p>
          </div>
        </div>
      </div>
    )
  };
}

