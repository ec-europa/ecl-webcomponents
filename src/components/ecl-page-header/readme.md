# ecl-page-header



<!-- Auto Generated Below -->


## Properties

| Property       | Attribute       | Description | Type     | Default     |
| -------------- | --------------- | ----------- | -------- | ----------- |
| `headerTitle`  | `header-title`  |             | `string` | `undefined` |
| `image`        | `image`         |             | `string` | `undefined` |
| `imageAlt`     | `image-alt`     |             | `string` | `undefined` |
| `meta`         | `meta`          |             | `string` | `undefined` |
| `overlay`      | `overlay`       |             | `string` | `undefined` |
| `styleClass`   | `style-class`   |             | `string` | `''`        |
| `theme`        | `theme`         |             | `string` | `'ec'`      |
| `thumbnail`    | `thumbnail`     |             | `string` | `undefined` |
| `thumbnailAlt` | `thumbnail-alt` |             | `string` | `undefined` |
| `variant`      | `variant`       |             | `string` | `'default'` |


## Dependencies

### Depends on

- [ecl-picture](../ecl-picture)

### Graph
```mermaid
graph TD;
  ecl-page-header --> ecl-picture
  style ecl-page-header fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
