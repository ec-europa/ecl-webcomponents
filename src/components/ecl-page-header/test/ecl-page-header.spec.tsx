import { newSpecPage } from '@stencil/core/testing';
import { EclPageHeader } from '../ecl-page-header';
import { EclBreadcrumb} from '../../ecl-breadcrumb/ecl-breadcrumb';
import { EclBreadcrumbItem} from '../../ecl-breadcrumb/ecl-breadcrumb-item';

describe('ecl-page-header', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclPageHeader, EclBreadcrumb, EclBreadcrumbItem],
      html: `<ecl-page-header
        variant="negative"
        theme="ec"
        header-title="Page title"
        image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image2.jpg"
        thumbnail="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image2.jpg"
        thumbnail-alt="Eu map"
        meta="Meta info | DD Month YYYY"
      >
        <ecl-breadcrumb
          slot="breadcrumb"
          variant="negative"
          theme="ec"
          ecl-script
         >
          <ecl-breadcrumb-item
            path="/example.html"
            variant="negative"
            theme="ec"
          >
            Home
          </ecl-breadcrumb-item>
          <ecl-breadcrumb-item
            style-class="ecl-breadcrumb__segment--ellipsis"
            ellipsis
            button-aria-label="Click to expand"
           >
           </ecl-breadcrumb-item>
          <ecl-breadcrumb-item
            variant="negative"
            path="/example.html"
            theme="ec"
          >
            About the European Commission
          </ecl-breadcrumb-item>
          <ecl-breadcrumb-item
            variant="negative"
            path="/example.html"
            theme="ec"
          >
            Organisational structure
          </ecl-breadcrumb-item>
          <ecl-breadcrumb-item
            variant="negative"
            path="/example.html"
            theme="ec"
          >
            How the Commission is organised
          </ecl-breadcrumb-item>
          <ecl-breadcrumb-item
            variant="negative"
            current-page
            style-class="ecl-breadcrumb__current-page"
            theme="ec"
          >
            News
          </ecl-breadcrumb-item>
        </ecl-breadcrumb>
        Lorem ipsum dolor sit amet, <a class="ecl-link" href="/example.html">consectetur adipiscing elit</a>.
        Quisque nec ullamcorper mi. Morbi interdum fermentum tempus. Nam nec rhoncus risus,
        <a class="ecl-link" href="/example.html">eget dictum elit</a>. Vestibulum gravida tincidunt venenatis
      </ecl-page-header>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
