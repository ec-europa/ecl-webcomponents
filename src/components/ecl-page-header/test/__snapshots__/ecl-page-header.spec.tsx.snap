// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-page-header renders 1`] = `
<ecl-page-header header-title="Page title" image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image2.jpg" meta="Meta info | DD Month YYYY" theme="ec" thumbnail="https://inno-ecl.s3.amazonaws.com/media/examples/example-image2.jpg" thumbnail-alt="Eu map" variant="negative">
  <!---->
  <div class="ecl-page-header ecl-page-header--image ecl-page-header--negative">
    <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image2.jpg" imgclass="ecl-page-header__background sc-ecl-page-header-ec" styleclass="ecl-page-header__picture-background sc-ecl-page-header-ec"></ecl-picture>
    <div class="ecl-container">
      <ecl-breadcrumb ecl-script="" slot="breadcrumb" theme="ec" variant="negative">
        <!---->
        <nav class="ecl-breadcrumb ecl-breadcrumb--negative ecl-page-header__breadcrumb sc-ecl-page-header-ec" data-ecl-breadcrumb>
          <ol class="ecl-breadcrumb__container sc-ecl-page-header-ec">
            <li class="ecl-breadcrumb__segment sc-ecl-breadcrumb-ec sc-ecl-page-header-ec" data-ecl-breadcrumb-item="static">
              <ecl-link path="/example" style-class="sc-ecl-breadcrumb-ec ecl-breadcrumb__link ecl-link--no-visited ecl-link--negative" variant="standalone">
                <ecl-icon icon="corner-arrow" size="2xs" slot="icon-after" style-class="ecl-breadcrumb__icon sc-ecl-breadcrumb-ec" transform="rotate-90"></ecl-icon>
                Home
              </ecl-link>
            </li>
            <li class="ecl-breadcrumb__segment ecl-breadcrumb__segment--ellipsis sc-ecl-breadcrumb-ec sc-ecl-page-header-ec" data-ecl-breadcrumb-ellipsis="data-ecl-breadcrumb-ellipsis" data-ecl-breadcrumb-item="static">
              <ecl-button aria-label="Click to expand" data-ecl-breadcrumb-ellipsis-button style-class="ecl-breadcrumb__ellipsis sc-ecl-breadcrumb-ec" variant="ghost">
                ...
                <ecl-icon icon="corner-arrow" size="2xs" slot="icon-after" style-class="ecl-breadcrumb__icon sc-ecl-breadcrumb-ec" transform="rotate-90"></ecl-icon>
              </ecl-button>
            </li>
            <li class="ecl-breadcrumb__segment sc-ecl-breadcrumb-ec sc-ecl-page-header-ec" data-ecl-breadcrumb-item="expandable">
              <ecl-link path="/example" style-class="sc-ecl-breadcrumb-ec ecl-breadcrumb__link ecl-link--no-visited ecl-link--negative" variant="standalone">
                <ecl-icon icon="corner-arrow" size="2xs" slot="icon-after" style-class="ecl-breadcrumb__icon sc-ecl-breadcrumb-ec" transform="rotate-90"></ecl-icon>
                About the European Commission
              </ecl-link>
            </li>
            <li class="ecl-breadcrumb__segment sc-ecl-breadcrumb-ec sc-ecl-page-header-ec" data-ecl-breadcrumb-item="expandable">
              <ecl-link path="/example" style-class="sc-ecl-breadcrumb-ec ecl-breadcrumb__link ecl-link--no-visited ecl-link--negative" variant="standalone">
                <ecl-icon icon="corner-arrow" size="2xs" slot="icon-after" style-class="ecl-breadcrumb__icon sc-ecl-breadcrumb-ec" transform="rotate-90"></ecl-icon>
                Organisational structure
              </ecl-link>
            </li>
            <li class="ecl-breadcrumb__segment sc-ecl-breadcrumb-ec sc-ecl-page-header-ec" data-ecl-breadcrumb-item="static">
              <ecl-link path="/example" style-class="sc-ecl-breadcrumb-ec ecl-breadcrumb__link ecl-link--no-visited ecl-link--negative" variant="standalone">
                <ecl-icon icon="corner-arrow" size="2xs" slot="icon-after" style-class="ecl-breadcrumb__icon sc-ecl-breadcrumb-ec" transform="rotate-90"></ecl-icon>
                How the Commission is organised
              </ecl-link>
            </li>
            <li class="ecl-breadcrumb__current-page ecl-breadcrumb__segment sc-ecl-breadcrumb-ec sc-ecl-page-header-ec" data-ecl-breadcrumb-item="static">
              News
            </li>
          </ol>
        </nav>
      </ecl-breadcrumb>
      <div class="ecl-page-header__meta">
        <span class="ecl-page-header__meta-item">
          Meta info | DD Month YYYY
        </span>
      </div>
      <div class="ecl-page-header__title-container">
        <h1 class="ecl-page-header__title">
          Page title
        </h1>
      </div>
      <div class="ecl-page-header__description-container">
        <img alt="Eu map" class="ecl-page-header__description-thumbnail" src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image2.jpg">
        <p class="ecl-page-header__description">
          Lorem ipsum dolor sit amet,
          <a class="ecl-link" href="/example">
            consectetur adipiscing elit
          </a>
          . Quisque nec ullamcorper mi. Morbi interdum fermentum tempus. Nam nec rhoncus risus,
          <a class="ecl-link" href="/example">
            eget dictum elit
          </a>
          . Vestibulum gravida tincidunt venenatis
        </p>
      </div>
    </div>
  </div>
</ecl-page-header>
`;
